migration-engine/migration-engine-tests/src/assertions.rs

```diff
@@ -51,15 +67,43 @@ impl SchemaAssertion {
     fn find_table<'a>(&'a self, table_name: &str) -> TableWalker<'a> {
         match self.find_table_option(table_name) {
             Some(table) => table,
-            None => panic!(
-                "assert_has_table failed. Table {} not found. Tables in database: {:?}",
-                table_name,
-                self.schema
-                    .table_walkers()
-                    .map(|table| table.name())
-                    .collect::<Vec<_>>()
+            None => self.assert_error(table_name, true),
+        }
+    }
+
+    // TODO: How can I instead return 'Void'?
```

Comment:
Not 100% sure that matches what you want, but you can give `!` as the return type to indicate that a function diverges (never returns, or panics, or aborts). So `fn assert_error(&self, table_name: &str, positive: bool) -> !`.